sno/working_copy/postgis.py

```diff
@@ -120,6 +141,20 @@ def session(self, bulk=0):
                 del self._db
                 L.debug("session: new/done")
 
+    def _register_types(self, db):
+        dbcur = db.cursor()
+        dbcur.execute("SELECT NULL::geometry")
+        geom_oid = dbcur.description[0][1]
+        GEOMETRY = psycopg2.extensions.new_type((geom_oid,), "GEOMETRY", cast_geometry)
+        psycopg2.extensions.register_type(GEOMETRY)
+
+        dbcur.execute("SELECT NULL::timestamp")
```

Comment:
technically these are all static values for builtin types: https://github.com/psycopg/psycopg2/blob/master/psycopg/typecast_builtins.c

(and there are a couple of flavours of timestamp, which one do we want here?)

Reply:
No idea why in postgres' datetime type is called TIMESTAMP, but that builtins.c calls it DATETIME.
Nevertheless, chose to apply this adapter-function to the non-timezone flavour (oid=1114) and the code works - since that is the only type i'm creating right now - GPKG DATETIMES are equivalent to Postgis TIMESTAMP, and right now there's no way to create TIMESTAMPTZs. Still have a TODO to fix this up later.